tests/test_objectives.py

```diff
@@ -7,12 +7,24 @@
 
 from symfit import (
     Variable, Parameter, Eq, Ge, Le, Lt, Gt, Ne, parameters, ModelError, Fit,
-    Model, FitResults, variables, CallableNumericalModel, Constraint
+    Model, FitResults, variables, CallableNumericalModel, Constraint, Idx,
+    IndexedBase, symbols, Sum, log
 )
 from symfit.core.objectives import (
     VectorLeastSquares, LeastSquares, LogLikelihood, MinimizeModel
 )
 from symfit.core.fit_results import FitResults
+from symfit.core.printing import SymfitNumPyPrinter
+from symfit.distributions import Exp
+
+# Overwrite the way Sum is printed by numpy just while testing. Is not
+# general enough to be moved to SymfitNumPyPrinter, but has to be used
+# in this test.
```

Comment:
Why does it have to be used?

Reply:
In order to build the analytical chi2 function we need to sum over `i`, but that is not printed currently by the numpy printer. And the solution here of replacing by `np.sum` is in general definitely not correct since that sums over all dimensions, not just that specified in `Sum`.

Reply:
You could do a sum by axis, but that'll be iffy to implement correctly I guess. Also, add that explanation to the comment